source/MaterialXRenderOsl/OslRenderer.cpp

```diff
@@ -410,13 +416,38 @@ void OslRenderer::render()
 
 ImagePtr OslRenderer::captureImage()
 {
-    // No-op: image capture is done as part of rendering.
-    return nullptr;
+    // As rendering goes to disk need to read the image back from disk
```

Comment:
Write implementations for vanilla OSL "capture" + "save" so we can get images back after rendering.